src/util/json.cpp

```diff
@@ -162,3 +163,44 @@ void jsont::swap(jsont &other)
   other.object.swap(object);
   other.value.swap(value);
 }
+
+bool operator==(const jsont &left, const jsont &right)
+{
+  if(left.kind != right.kind)
+    return false;
+  switch(left.kind)
+  {
+  case jsont::kindt::J_NULL:
+    return true;
+  case jsont::kindt::J_TRUE:
+    return true;
+  case jsont::kindt::J_FALSE:
+    return true;
+  case jsont::kindt::J_NUMBER:
+    return left.value == right.value;
+  case jsont::kindt::J_STRING:
+    return left.value == right.value;
+  case jsont::kindt::J_ARRAY:
+  {
+    const auto &left_array = static_cast<const json_arrayt &>(left);
```

Comment:
:information_source: There is a function called `to_json_array` at the bottom of this file, which you could re-use instead of adding your own static casts.

Reply:
`static_cast` is a feature of C++, and not "my" cast. `to_json_array` checks for kind, which in this case is already checked. We'd be doing extra work, so no.